### PR TITLE
[MODULAR] Borer tweaks - no more free lunch.

### PR DESCRIPTION
**Pull request by Higgin**
## About The Pull Request

Most borer abilities now cause brain damage - including leveling up, learning new chems, injecting chems, and making eggs (which now has a chance to directly cause severe traumas.)
Borer passive tox gain goes much higher than it did before - high enough to be an actual pain to deal with umanaged.
Borers can no longer pump out brain damage/trauma-fixing chems (Mannitol and Neurine.)
Borers no longer get jacked-up move/actionspeed on top of table passthrough and hiding. 
Borers must be in LIVING hosts to heal, regain chems, or gain stat/chem evolution points.
Borers entering hosts now have a slightly tighter margin for doing so (6 second do_after versus 7 second paralyze - act quick.)

Borer team objectives are now much more realistic and achievable.
Willing hosts give a much larger relative boost to the speed of borer levelups after you get enough of them.
Fearing hosts/potential victims now causes stamina damage and confusion which briefly outlasts the paralyze (combine with haloperidol for particularly unruly hosts.)
Borers now also have access to Multiver, a basic chem/tox purger, without having to learn it from blood - and Mindbreaker Toxin (which can be used to induce hallucinations as well as addiction to hallucinogens - a possible path to actually brainwashing a host when they start to withdraw.)

## How This Contributes To The Skyrat Roleplay Experience

Hosts now actually have a reason to fear getting pumped and dumped by borers; borers now have a realistic expectation of achieving their objectives whether they pursue them cooperatively or coercively; borers and hosts looking to work together must now handle the need for brain healing separate from the borer's ability to learn/inject mannitol and neurine.

This makes borers a much more interesting midround. With the right chemical management, and particularly if they act passively as a focus for their host, they're still highly-beneficial allies and clinch hitters with a good array of healing chems (and the ability to learn basically any synthesizable chem except the brain healers.) 

However, in order to advance and pursue their own objectives, they must hurt their own hosts, negotiate the much more pernicious costs to keeping them well, and court greater danger when outside of a host. They can benefit a lot more from willing hosts now. The benefits are real but so are the tradeoffs for both parties.

TANSTAAFL

## Changelog

:cl:
balance: Major borer rebalance. Borer abilities now generally cause brain damage up to and including traumas. Borers can no longer heal brain damage/traumas on their own. Borers require living hosts to gain stats, chems, or heal. Hosts must manage more significant costs even with passive borers. Borer objectives are now much more realistically achievable, and willing hosts contribute to much greater boosts for borers than unwilling hosts. Borer actionspeed/movespeed buffs reduced. Borers have stamina damage and confusion on their Fear ability as well as the ability to make Multiver (a chem purger) and Mindbreaker Toxin (a hallucinogen that can be used to brainwash/incapacitate) without learning from prior samples. 
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 1]
```diff
@@ -1,14 +1,14 @@
-GLOBAL_VAR_INIT(objective_egg_borer_number, 5)
-GLOBAL_VAR_INIT(objective_egg_egg_number, 10)
-GLOBAL_VAR_INIT(objective_willing_hosts, 10)
-GLOBAL_VAR_INIT(objective_blood_chem, 5)
-GLOBAL_VAR_INIT(objective_blood_borer, 5)
+GLOBAL_VAR_INIT(objective_egg_borer_number, 2)
+GLOBAL_VAR_INIT(objective_egg_egg_number, 5)
+GLOBAL_VAR_INIT(objective_willing_hosts, 2)
+GLOBAL_VAR_INIT(objective_blood_chem, 3)
+GLOBAL_VAR_INIT(objective_blood_borer, 3)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:5]: uh oh, here begins more powercreep! (love it)

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 2]
```diff
@@ -289,12 +294,12 @@ GLOBAL_VAR_INIT(successful_blood_chem, 0)
 
 /mob/living/simple_animal/cortical_borer/Life(delta_time, times_fired)
 	. = ..()
-	//can only do stuff when we are inside a human
-	if(!inside_human())
+	//can only do stuff when we are inside a LIVING human
+	if(!inside_human() || (human_host.stat == DEAD))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: you don't need the parenthesis around the second arg. you do however need to make sure there is a human host, because it will runtime with "null.stat" if there is no host.

Comment (jjpark-kb) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: so maybe human_host?.stat == DEAD, or something like that

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 3]
```diff
@@ -289,12 +294,12 @@ GLOBAL_VAR_INIT(successful_blood_chem, 0)
...
 		return
 
 	//there needs to be a negative to having a borer
-	if(prob(5) && human_host.getToxLoss() <= 20)
+	if(prob(5) && human_host.getToxLoss() <= 80)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:306]: this is brutal because remember, whenever someone has a borer, they are unable to see their health (similar to miner salve).

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 4]
```diff
@@ -324,11 +329,11 @@ GLOBAL_VAR_INIT(successful_blood_chem, 0)
...
 		var/maturity_threshold = 20
 		if(GLOB.successful_egg_number >= GLOB.objective_egg_borer_number)
-			maturity_threshold -= 5
+			maturity_threshold -= 3
 		if(length(GLOB.willing_hosts) >= GLOB.objective_willing_hosts)
-			maturity_threshold -= 5
+			maturity_threshold -= 10
 		if(GLOB.successful_blood_chem >= GLOB.objective_blood_borer)
-			maturity_threshold -= 5
+			maturity_threshold -= 3
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: originally, it would add up to 15 being removed-- here it removes 16; so, you may need to lessen on of the things. Maybe the 10 to a 9, or one of the 3s to a 2?

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 5]
```diff
@@ -32,7 +32,10 @@
 		return
 	cortical_owner.reagent_holder.reagents.add_reagent(choice, 5, added_purity = 1)
 	cortical_owner.reagent_holder.reagents.trans_to(cortical_owner.human_host, 30, methods = INGEST)
-	to_chat(cortical_owner.human_host, span_warning("You feel something cool inside of you!"))
+	var/obj/item/organ/internal/brain/victim_brain = cortical_owner.human_host.getorganslot(ORGAN_SLOT_BRAIN)
+	if(victim_brain)
+		cortical_owner.human_host.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10)
+	to_chat(cortical_owner.human_host, span_warning("You feel something cool inside of you and a dull ache in your head!"))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: oh brain damage is rough... but sure?

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 6]
```diff
@@ -137,15 +140,23 @@
 		to_chat(owner, span_warning("You already know this chemical!"))
 		cortical_owner.chemical_evolution += 5
 		return
+	if(locate(reagent_choice) in cortical_owner.blacklisted_chemicals)
+		to_chat(owner, span_warning("Your physiology is incompatible with this chemical - find it elsewhere!"))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: remove the "find it elsewhere" since they can't learn it whatsoever-- which I'm still against blacklisting chems (beyond non-synthesizable ones).

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 7]
```diff
@@ -161,15 +161,20 @@ GLOBAL_VAR_INIT(successful_blood_chem, 0)
...
-										/datum/reagent/medicine/mannitol,
 										/datum/reagent/drug/methamphetamine/borer_version,
 										/datum/reagent/medicine/morphine,
 										/datum/reagent/medicine/inacusiate,
 										/datum/reagent/medicine/oculine,
+										/datum/reagent/toxin/mindbreaker,
+	)
+	//blacklisted chemicals - separate from chemicals that cannot be synthesized, borers specifically cannot learn these
+	var/list/blacklisted_chemicals = list(/datum/reagent/medicine/mannitol,
+										/datum/reagent/medicine/neurine,
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:177]: I'm against creating a blacklist whatsoever-- I'll live and we can see how it goes.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 8]
```diff
@@ -137,15 +140,23 @@
...
 		return
 	cortical_owner.known_chemicals += reagent_choice.type
 	cortical_owner.blood_chems_learned++
+	var/obj/item/organ/internal/brain/victim_brain = cortical_owner.human_host.getorganslot(ORGAN_SLOT_BRAIN)
+	if(victim_brain)
+		cortical_owner.human_host.adjustOrganLoss(ORGAN_SLOT_BRAIN, 5)
 	if(cortical_owner.blood_chems_learned == 5)
 		GLOB.successful_blood_chem += 1
 	to_chat(owner, span_notice("You have learned [initial(reagent_choice.name)]"))
+	to_chat(cortical_owner.human_host, span_notice("You get the strange aftertaste of [initial(reagent_choice.taste_description)]!"))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: I think changing "the" to "a" would make it more grammatically correct?

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 9]
```diff
@@ -184,7 +195,11 @@
 		return
 	cortical_owner.known_chemicals += reagent_choice
 	cortical_owner.potential_chemicals -= reagent_choice
+	var/obj/item/organ/internal/brain/victim_brain = cortical_owner.human_host.getorganslot(ORGAN_SLOT_BRAIN)
+	if(victim_brain)
+		cortical_owner.human_host.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10)
 	to_chat(owner, span_notice("You have learned [initial(reagent_choice.name)]"))
+	to_chat(cortical_owner.human_host, span_notice("You get the strange aftertaste of [initial(reagent_choice.taste_description)]!"))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: same as the above comment about "the" to "a"

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 10]
```diff
@@ -279,6 +301,8 @@
 		var/mob/living/carbon/human/singular_fear = pick(potential_freezers)
 		to_chat(singular_fear, span_warning("Something glares menacingly at you!"))
 		singular_fear.Paralyze(7 SECONDS)
+		singular_fear.adjustStaminaLoss(50)
+		singular_fear.set_timed_status_effect(9 SECONDS, /datum/status_effect/confusion, only_if_higher = TRUE)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:289]: this is all very brutal-- I like it.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 11]
```diff
@@ -321,6 +347,9 @@
 	if(!cortical_owner.human_host)
 		to_chat(owner, span_warning("You must have a host to check blood!"))
 		return
+	var/obj/item/organ/internal/brain/victim_brain = cortical_owner.human_host.getorganslot(ORGAN_SLOT_BRAIN)
+	if(victim_brain)
+		cortical_owner.human_host.adjustOrganLoss(ORGAN_SLOT_BRAIN, 1)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: mm, I don't think doing brain damage is good for this one... maybe remove some blood?

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 12]
```diff
@@ -680,7 +709,7 @@
 		if(singular_host.has_borer())
 			to_chat(cortical_owner, span_warning("You cannot occupy a body already occupied!"))
 			return
-		if(!do_after(cortical_owner, 5 SECONDS, target = singular_host))
+		if(!do_after(cortical_owner, 6 SECONDS, target = singular_host))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:695]: I only set it to 5 because lag-- this might be fine, we will have to see.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 13]
```diff
@@ -789,9 +821,18 @@
...
+	var/obj/item/organ/internal/brain/victim_brain = cortical_owner.human_host.getorganslot(ORGAN_SLOT_BRAIN)
+	if(victim_brain)
+		cortical_owner.human_host.adjustOrganLoss(ORGAN_SLOT_BRAIN, 25)
+		if(prob(33))
+			cortical_owner.human_host.gain_trauma_type(BRAIN_TRAUMA_MILD, TRAUMA_RESILIENCE_BASIC)
+		else if(prob(25))
+			cortical_owner.human_host.gain_trauma_type(BRAIN_TRAUMA_MILD, TRAUMA_RESILIENCE_SURGERY)
+		else if(prob(10))
+			cortical_owner.human_host.gain_trauma_type(BRAIN_TRAUMA_SEVERE, TRAUMA_RESILIENCE_SURGERY)
+		else if(prob(3))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: yandere simulator moment-- a switch from a rand(1,100) might be better, I have no clue.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 14]
```diff
@@ -915,7 +956,7 @@
 
 //to ask if a host is willing
 /datum/action/cooldown/willing_host
-	name = "Willing Host (300 chemicals)"
+	name = "Willing Host (150 chemicals)"
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:949]: oh wowee, that is a huge buff.

**Comment by nevimer** [comment]
Borers already have the issue of being a semi frequent roll that nobody hits yes for. I'd imagine making them more difficult to play would worsen that. And borers killing everyone via brain damage is incredibly lame to be victim of, let alone be forced to do so as the borer. Players here will actually cryo if you go into them and cause traumas, from my experiences. 

almost every time I got into someone in the past two months they B-lined for sugar, screamed borer, shit their pants, and spammed medicals door. A certain someone will doubt this happened, but I don't give a shit to go prove things. 

This is how they should be re-balanced. Their goals should be redone, they should be interesting and antagonistic. The REAL issue with borers isn't their mechanical functions, if anything, the tox is metagay and anyone who knows anything will just say borer, and next thing you know you're a waste of threat that could have been used for a role that actually causes danger. 

Here is how to actually fix borers:

Insist they make their hosts do EVIL things. Make the hosts have to steal things, say certain thing, infiltrate certain people. Borers are already weak as shit mechanically to anyone who knows to turn on combat mode and walk into them man, some seriously unrobust fuckers should not be affecting their balance. 

There are all sorts of stupid metagamey methods to kill borers the general mass of players have not caught onto yet, such as stasis and sugar. Gay baby jail, effectively. Heating the pipenets kills them. Cooling them kills them. Literally doing anything kills them. Don't make them squishy, they're already a bigger waste of threat than anything else, a non-event, even. 

And willing host? By the time you have it, 300, 150 chemicals, doesn't even matter. You're a chem dispenser and by the time the ability is back you have already generated like 100x what it would cost to use. You generate chemicals faster than you can click at even level one. "But I just tested it and--" shut the fuck up, when are you giving a dude literally as fast as you can click levels of any chemical you can learn right off the bat lol. 

If you still insist on mechanical changes, perhaps, remove the meth. Don't remove the mannitol. Make some unique chems for the borer with new abilities to AID in antagonizing instead of running fast for no reason. Make the victim a "brainwashed victim" or some shit. MAKE THEM INTERESTING. The only people who give two fucks about a person with a borer are security, and security. And why? "borer bad". 

tl;dr make borers brainwash victims as an ability or some shit && give them real goals other than to spread and survive. make them an actual antagonist.

I'll even do this myself if I have to.

**Comment by Higgin** [comment]
> Insist they make their hosts do EVIL things. Make the hosts have to steal things, say certain thing, infiltrate certain people. Borers are already weak as shit mechanically to anyone who knows to turn on combat mode and walk into them man, some seriously unrobust fuckers should not be affecting their balance.

Honestly, I'd like to do this, but I'm not practically sure how. I also like borers having a dual nature - they CAN help you out. They're a deuteragonist - they can go both ways. The problem I'm trying to address is that their 'antagonism' in the pursuit of their regular survive/reproduce goals had no real tradeoff or drawback. It was basically only beneficial to take a borer if they remotely knew what they were doing. This way, there are tradeoffs to taking them - there are reasons to NOT take them - but they can also offer you a lot of boosts and a useful companion if you manage those tradeoffs well (an MR virus, asking for some mannitol/neurine which is dogshit easy to make since the purity changes both come to mind.)

Making it so they can ONLY have an antagonistic character wouldn't be as interesting either. If I were to do it, I was thinking I'd like some borers to have the more antagonistic goals for manipulating their hosts into stealing/killing etc. - others not. It's more interesting when there's variety. I'd welcome you doing it if you're really up for it @nevimer 

`also, for what it's worth, i don't really give a shit about borers being mechanically 'weak as shit' to anyone who knows how to throw floor tiles at a table or get a medHUD, midrounds were not born to necessarily live long and happy lives. if you can actually get people superheating the pipenet for you, you're doing God's work. if they're eating up too much threat too commonly in dynamic for a midround spawn that often doesn't go far (when we have a lot of other midrounds already disabled, I feel this too) that's a separate issue. `

> And willing host? By the time you have it, 300, 150 chemicals, doesn't even matter. You're a chem dispenser and by the time the ability is back you have already generated like 100x what it would cost to use. You generate chemicals faster than you can click at even level one. "But I just tested it and--" shut the fuck up, when are you giving a dude literally as fast as you can click levels of any chemical you can learn right off the bat lol.

the point of nerfing the willing host chemical cost is that you have to evolve inside a host multiple times to get the chemical pool up to where you can actually offer willing host (takes five Become Strongers - 50->150) with these changes, it means you can much earlier offer willing host (opening up access to the big buffs you get by completing the objective) while now actually having had to make your hosts suffer a bit to make the offer.

**Comment by nevimer** [comment]
Borer families. Hot idea.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm (hunk 2 above)
Comment (Higgin) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: Oddly enough it wasn't runtiming while outside of the host, I think inside_human might have caught that - but I changed it to human_host?.stat == DEAD and it seems fine.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm (hunk 4 above)
Comment (Higgin) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: Dropped the egg goal to a 2 since it could practically be farmed on monkeys - brings everything to 15 again.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm (hunk 11 above)
Comment (Higgin) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: Set it to create a small blood drip. It's an easily-spammed ability that you might easily want to spam to keep an eye on host status - brain damage wasn't good for this one.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm (hunk 13 above)
Comment (Higgin) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]: drank from the neurine chalice and changed it to a switch

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm (hunk 14 above)
Comment (Higgin) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:949]: 5 evolutions before you can get enough chems to even afford it means at least 125 brain damage and whatever trouble comes with that.

**Comment by nevimer** [comment]
Time to see more cortical borer rolls fail to spawn lol
